src/java.desktop/share/classes/javax/swing/plaf/basic/BasicScrollBarUI.java

```diff
@@ -355,12 +360,15 @@ protected void installListeners(){
         buttonListener = createArrowButtonListener();
         modelListener = createModelListener();
         propertyChangeListener = createPropertyChangeListener();
+        keyboardFocusManager = KeyboardFocusManager.getCurrentKeyboardFocusManager();
+        keyboardFocusListener = createKeyboardFocusListener();
 
         scrollbar.addMouseListener(trackListener);
         scrollbar.addMouseMotionListener(trackListener);
         scrollbar.getModel().addChangeListener(modelListener);
         scrollbar.addPropertyChangeListener(propertyChangeListener);
         scrollbar.addFocusListener(getHandler());
+        keyboardFocusManager.addPropertyChangeListener(keyboardFocusListener);
```

Comment:
creating `keyboardFocusManager` instance can be avoided and we can directly use `KeyboardFocusManager.getCurrentKeyboardFocusManager().addPropertyChangeListener(keyboardFocusListener);` to add the listener and similarly to remove.